client/flutter/lib/pages/home_page.dart

```diff
@@ -123,4 +138,28 @@ class _HomePageState extends State<HomePage> {
       ),
     );
   }
+
+  Future<void> _tryShareOrShowSnackBar(BuildContext context) async {
+    const url = 'https://www.who.int/covid-19-app';
+
+    if (!kIsWeb && (Platform.isAndroid || Platform.isIOS)) {
+      await Share.share(S.of(context).commonWhoAppShareIconButtonDescription);
+    } else {
+      final snackBar = SnackBar(
+        backgroundColor: Constants.primaryColor,
+        duration: const Duration(seconds: 6),
+        action: SnackBarAction(
+          label: 'Open link',
+          onPressed: () async => await launch(url),
+        ),
+        content: Text('Link copied to clipboard.'),
+      );
+
+      await Clipboard.setData(ClipboardData(
```

Comment:
Reorder to start with copying before saying we copied

Reply:
@theswerd The `SnackBar` build before copying to clipboard is only shown after the copying. See the line below.

I'm rebasing to fix conflicts as we speak.